openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -261,3 +267,69 @@ def post(self, request, usage_key_str):
         except ValueError as err:
             raise ValidationError(detail=str(err))
         return Response(LibraryXBlockOlxSerializer({"olx": new_olx_str}).data)
+
+
+@view_auth_classes()
+class LibraryBlockAssetListView(APIView):
+    """
+    Views to list an existing XBlock's static asset files
+    """
+    @convert_exceptions
+    def get(self, request, usage_key_str):
+        """
+        List the static asset files belonging to this block.
+        """
+        key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        files = api.get_library_block_static_asset_files(key)
+        return Response(LibraryXBlockStaticFilesSerializer({"files": files}).data)
+
+
+@view_auth_classes()
+class LibraryBlockAssetView(APIView):
+    """
+    Views to work with an existing XBlock's static asset files
+    """
+    parser_classes = (MultiPartParser, )
+
+    @convert_exceptions
+    def get(self, request, usage_key_str, file_path):
+        """
+        Get a static asset file belonging to this block.
+        """
+        key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        files = api.get_library_block_static_asset_files(key)
+        for f in files:
+            if f.path == file_path:
+                return Response(LibraryXBlockStaticFileSerializer(f).data)
+        raise NotFound
+
+    @convert_exceptions
+    def put(self, request, usage_key_str, file_path):
+        """
+        Replace a static asset file belonging to this block.
+        """
+        usage_key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        file_wrapper = request.data['content']
+        if file_wrapper.size > 20 * 1024 * 1024:  # > 20 MiB
+            # In the future, we need a way to use file_wrapper.chunks() to read
+            # the file in chunks and stream that to Blockstore, but Blockstore
+            # currently lacks an API for streaming file uploads.
+            raise ValidationError("File too big")
+        file_content = file_wrapper.read()
+        try:
+            result = api.add_library_block_static_asset_file(usage_key, file_path, file_content)
+        except ValueError:
+            raise ValidationError("Invalid file path")
+        return Response(LibraryXBlockStaticFileSerializer(result).data)
+
+    @convert_exceptions
+    def delete(self, request, usage_key_str, file_path):  # pylint: disable=unused-argument
+        """
+        Delete a static asset file belonging to this block.
+        """
+        usage_key = LibraryUsageLocatorV2.from_string(usage_key_str)
+        try:
+            api.delete_library_block_static_asset_file(usage_key, file_path)
+        except ValueError:
+            raise ValidationError("Invalid file path")
+        return Response({})
```

Comment:
Minor, but I think DRF defaults to a 204 and no content for deletes and not 200 and empty JSON dict.

Reply:
Sure, changed to a 204 response.